compiler/src/dotty/tools/dotc/core/JavaNullInterop.scala

```diff
@@ -95,8 +98,17 @@ object JavaNullInterop {
    *                                       This is useful for e.g. constructors, and also so that `A & B` is nullified
    *                                       to `(A & B) | Null`, instead of `(A | Null & B | Null) | Null`.
    */
-  private class JavaNullMap(var outermostLevelAlreadyNullable: Boolean)(using Context) extends TypeMap {
-    def nullify(tp: Type): Type = if ctx.flexibleTypes then FlexibleType(tp) else OrNull(tp)
+  private class JavaNullMap(var outermostLevelAlreadyNullable: Boolean, explicitlyNullable: Boolean)(using Context) extends TypeMap {
```

Comment:
I don't think the `NotNull`/`Nullable` annotation on the symbol should affect any types nested inside.

Reply:
I'm not sure what you mean by this, I add the explicitlyNullable flag here to distinguish the case between a flexible type and an OrNull. Are you referring to the places where we call `this` in `apply`?